### ISSUE TITLE
Status pages serves 503

### ISSUE THREAD

**Issue by CyBeRoni** [zotonic/zotonic#1243]
As of a01e458b1590431786501c101731ac414521121d, the status page is served with a 503, causing my monitoring to think it's broken.

I understand the reasoning of sending a 503 for that page if it is served as the fallback mechanism for a site (and I agree with that idea), however if the site is served as a result of a a specific request the correct code to serve is 200.

I propose that the status site does not send status 503 if the hostname as requested is the same as the hostname configured for the status site, if the hostname as requested is identical to the server hostname, or if no Host header is set.

**Comment by mworrell** [comment]
I agree that 503 is not always the correct error here.

We might want to return:

  * 503: for hostnames of all non-running (but retrying) sites
  * 404: for all unknown hostnames (also for disabled sites)
  * 200: if no Host header is present

In addition we should ensure that the *no-index* header is set.

**Comment by CyBeRoni** [comment]
> 200: if no Host header is present

Or if explicitly requested as I proposed above. Various browsers have in the past also pulled shenanigans if served an error status, replacing the page with their own error message. Such behaviour would make the status page entirely unusable since there is also no way that I know of to force a browser not to send a Host header.

The zotonic_status site does have a setting for hostname, though it is set to none and is apparently unused. And again I think if the Host header is equal to the system hostname, it is also not unreasonable to assume the intended result was to get the status site.

**Comment by ddeboer** [comment]
@arjan Do you have time to work on this? We may want to fix this on master (Cowboy) only.

**Comment by mworrell** [comment]
I agree with @CyBeRoni that if the requested `Host` is the system name, that we can assume that the zotonic_status page is directly requested.

I assign this to myself.

**Comment by CyBeRoni** [comment]
Yes, though an identically configured site name should override that (so people who name their system 'www.somesite.com' don't get into trouble).

**Comment by CyBeRoni** [comment]
I also still think it would be useful to use the site name as configured as well, so that one can make the choice of where to find the status site explicit.

**Comment by mworrell** [comment]
@CyBeRoni Is this still an issue, or was it already fixed for you in the 1.0?
